src/read.c

```diff
@@ -1087,58 +1087,44 @@ static avifResult avifDecoderItemValidateProperties(const avifDecoderItem * item
     }
 
     if (!memcmp(item->type, "grid", 4)) {
-        // MIAF (ISO 23000-22:2019), Section 7.3.11.4.1:
-        //   All input images of a grid image item shall use the same coding format, chroma sampling format, and the
-        //   same decoder configuration (see 7.3.6.2).
-        avifDecoderItem * firstTile = NULL;
         for (uint32_t i = 0; i < item->meta->items.count; ++i) {
             avifDecoderItem * tile = &item->meta->items.item[i];
-            if (tile->dimgForID == item->id) {
-                // Skip the same items as avifDecoderGenerateImageGridTiles().
-                if (avifGetCodecType(item->type) == AVIF_CODEC_TYPE_UNKNOWN) {
-                    continue;
-                }
+            if (tile->dimgForID != item->id) {
+                continue;
+            }
+            // Tile item types were checked in avifDecoderGenerateImageTiles(), no need to do it here.
```

Comment:
Yannis: The function that checks tile item types is `avifDecoderGenerateImageGridTiles`. Since `avifDecoderGenerateImageTiles` calls `avifDecoderGenerateImageGridTiles`, this comment is still correct. But I was wondering if you meant to say "avifDecoderGenerateImageGridTiles" here.

Reply:
Thanks for the confirmation. I thought that might be why you used "avifDecoderGenerateImageTiles" in the comment and wanted to confirm it with you.